### PR TITLE
test: cover AsyncMDNSResolver cache-only mdns_timeout behavior

### PR DESCRIPTION
**Pull request by aiolibsbot**
## What
Add regression tests for `AsyncMDNSResolver`'s cache-only behavior when `mdns_timeout` is falsy (`0` or `None`).

## Why
`docs/api.rst` documents that a `0`/`None` `mdns_timeout` "will only use the cache and will not perform a new query on the network." Only `AsyncDualMDNSResolver` had a zero-timeout test (`test_different_results_async_dual_mdns_resolver_zero_timeout`); the single resolver's contract was asserted only indirectly, so a regression that started issuing a network query (e.g. changing `if self._mdns_timeout:` to always call `async_request`) would slip through.

## How
Three tests on the `AsyncMDNSResolver` `resolve()` path, each asserting `async_request` is never awaited:
- cache **hit** → returns the cached address (no network query)
- cache **miss**, `mdns_timeout=0` → raises `OSError("MDNS lookup failed")` (no network query)
- cache **miss**, `mdns_timeout=None` → same, covering the `None` branch

## Testing
`python -m py_compile tests/test_impl.py` passes. The suite can't run in this environment (no `zeroconf`/`aiohttp`), but the tests follow the existing fixtures and patch patterns (`make_resolvers_patchable`, `IPv4HostResolver`, the `_mdns_timeout = 0` override used by the dual-resolver zero-timeout test) — CI will exercise them across the full matrix.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
### Quality Report

**Changes**: 2 files changed, 77 insertions(+)

**Code scan**: clean

**Tests**: failed (FAILED)

**Branch hygiene**: clean

*Generated by Kōan post-mission quality pipeline*

### CODE REVIEW

File: tests/test_impl.py [hunk 1]
```diff
@@ -273,6 +273,80 @@ async def test_resolve_mdns_name_af_inet6(resolver: AsyncMDNSResolver) -> None:
...
+async def test_resolve_mdns_name_zero_timeout_returns_cached(
+    resolver: AsyncMDNSResolver,
+) -> None:
+    """Test mdns_timeout=0 returns a cached result without a network query.
+
+    A falsy ``mdns_timeout`` (``0`` or ``None``) restricts the resolver to the
+    zeroconf cache, so a cache hit must be served without issuing an mDNS
+    request on the network.
+    """
+    resolver._mdns_timeout = 0
```
Comment (bdraco) [inline tests/test_impl.py]: It would be better to test these though the public api. maybe make resolver fixture a factory pattern

**Review by bdraco — CHANGES_REQUESTED** [review]
see inline review comments

**Comment by bdraco** [comment]
@aiolibsbot rr

File: tests/test_impl.py (hunk 1 above)
Comment (aiolibsbot) [reply tests/test_impl.py]: Agreed — constructing via the public `AsyncMDNSResolver(mdns_timeout=0)` is more robust than mutating the private `_mdns_timeout`, since the latter couples the test to an internal attribute name. A factory fixture keeps the per-test `close()` cleanup while letting each test pick the timeout through the public constructor:

```python
@pytest_asyncio.fixture
async def mdns_resolver() -> AsyncGenerator[Callable[..., AsyncMDNSResolver]]:
    resolvers: list[AsyncMDNSResolver] = []

    def _factory(**kwargs: Any) -> AsyncMDNSResolver:
        kwargs.setdefault("mdns_timeout", 0.1)
        resolver = AsyncMDNSResolver(**kwargs)
        resolvers.append(resolver)
        return resolver

    yield _factory
    for resolver in resolvers:
        await resolver.close()
```

The cache-miss tests then become `resolver = mdns_resolver(mdns_timeout=0)` / `mdns_resolver(mdns_timeout=None)` with no `_mdns_timeout` poke.

Two related notes:
1. The existing `test_different_results_async_dual_mdns_resolver_zero_timeout` mutates `_mdns_timeout` the same way (around line 609); a parallel `dual_resolver` factory could migrate it for consistency, if you'd like that cleaned up in the same PR.
2. For the cache-*hit* test specifically, the timeout value is irrelevant: on a cache hit `resolve()` returns from the `load_from_cache` branch (`_impl.py:149-150`) before `_resolve_mdns` — the only awaiter of `async_request` (`_impl.py:111-112`) — ever runs, so `async_request` is never called regardless of timeout. That test is really asserting the cache-hit short-circuit (already covered by `test_resolve_mdns_name_unspec_from_cache`), so it can be built with the default timeout or renamed; the genuine zero-timeout contract is what the two cache-miss tests verify.

**Review by bdraco — APPROVED** [review]
